client/state/data-layer/wpcom/akismet-key/index.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+	AKISMET_KEY_REQUEST,
+	AKISMET_KEY_RECIEVE,
+	AKISMET_KEY_REQUEST_FAILURE,
+} from 'calypso/state/action-types';
+import { registerHandlers } from 'calypso/state/data-layer/handler-registry';
+import { http } from 'calypso/state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'calypso/state/data-layer/wpcom-http/utils';
+
+const fetchAkismetAPIKey = ( action: object ) => {
+	return http(
```

Comment:
It's not a big deal but all this state could probably be implemented in just a handful of lines using `useQuery` instead of Redux actions, selectors, reducer, and the data-layer. It would be a lot less code and might have a performance benefit in addition to getting a number of features for free. (To my knowledge, React Query is how we're supposed to be adding state going forward.)

Reply:
Thanks @sirbrillig! TIL about `useQuery`. It was a quick refactor, so I went ahead and updated. Only 3 files in this PR now instead of 11 :)

Reply:
Nice work!